### PR TITLE
Added selected issue location highlighting

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Relates to #1597

### CODE REVIEW

File: src/IssueViz.UnitTests/Editor/Common/TaggerTestHelper.cs [hunk 1]
```diff
@@ -34,6 +35,9 @@ internal static class TaggerTestHelper
         public static ITextSnapshot CreateSnapshotAndBuffer(int length = 999) =>
             CreateBufferMockWithSnapshot(length).Object.CurrentSnapshot;
 
+        public static ITextBuffer CreateBufferWithSnapshot(int length = 999) =>
```
Comment (duncanp-sonar) [inline src/IssueViz.UnitTests/Editor/Common/TaggerTestHelper.cs:38]: Convenience method - sometimes the caller is more interested in the buffer, sometimes the snapshot.

File: src/IssueViz/Editor/SelectedIssueTagging/Highlighting/IssueHighlightViewTaggerProvider.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
+            }
+
+            if (buffer != textView.TextBuffer)
+            {
+                return null;
+            }
```
Comment (duncanp-sonar) [inline src/IssueViz/Editor/SelectedIssueTagging/Highlighting/IssueHighlightViewTaggerProvider.cs:59]: I don't know if we need to be this defensive, but I guess it doesn't hurt.

File: src/IssueViz/Editor/SelectedIssueTagging/Highlighting/IssueHighlightViewTaggerProvider.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+using Microsoft.VisualStudio.Utilities;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Editor.SelectedIssueTagging.Highlight
+{
+    [Export(typeof(IViewTaggerProvider))]
+    [ContentType("code")]
+    [TextViewRole(PredefinedTextViewRoles.PrimaryDocument)]
```
Comment (rita-g-sonarsource) [inline src/IssueViz/Editor/SelectedIssueTagging/Highlighting/IssueHighlightViewTaggerProvider.cs:32]: In TaggerProvider we used TextViewRole `Document` instead of `PrimaryDocument`. I see that primary document is `Applies to text views of documents that are open for mainline editing, excluding auxiliary views of documents.` -- I wonder what they mean by auxiliary views; peek definition for example? Shame that the documentation is so vague.

File: src/IssueViz.UnitTests/Editor/Common/TaggerTestHelper.cs (hunk 1 above)
Comment (rita-g-sonarsource) [reply src/IssueViz.UnitTests/Editor/Common/TaggerTestHelper.cs:38]: The method names are really confusing, `CreateSnapshotAndBuffer` vs `CreateBufferWithSnapshot`. 
Maybe just CreateSnapshot and CreateBuffer?

File: src/IssueViz/Editor/SelectedIssueTagging/Highlighting/IssueHighlightViewTaggerProvider.cs (hunk 3 above)
Comment (duncanp-sonar) [reply src/IssueViz/Editor/SelectedIssueTagging/Highlighting/IssueHighlightViewTaggerProvider.cs:32]: Yep, not sure. We can try different combinations and see if we can limit it to "real" editable documents. 
I think comparison views are another type of auxiliary document.

File: src/IssueViz/Editor/SelectedIssueTagging/Highlighting/IssueHighlightTag.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    [Export(typeof(EditorFormatDefinition))]
+    [Name(FormatName)]
+    [UserVisible(true)]
+    internal class IssueHighlightFormatDefinition : MarkerFormatDefinition
+    {
+        public const string FormatName = "MarkerFormatDefinition/SLVS_IssueHighlightFormatDefinition";
+
+        public IssueHighlightFormatDefinition()
+        {
+            BackgroundColor = Colors.LightPink;
```
Comment (rita-g-sonarsource) [inline src/IssueViz/Editor/SelectedIssueTagging/Highlighting/IssueHighlightTag.cs:46]: If you'll make another commit - can you please either add "this" to BackgroundColor, or remove it from the rest? 😄

**Review by rita-g-sonarsource — APPROVED** [review]
